src/joint.jl

```diff
@@ -348,6 +350,62 @@ function velocity_to_configuration_derivative!(q̇::AbstractVector, joint::Joint
     velocity_to_configuration_derivative!(q̇, joint.joint_type, q, v)
 end
 
+
+"""
+$(SIGNATURES)
+
+Compute the jacobian ``J_{v \\rightarrow \\dot{q}}`` which maps joint velocity to configuration
+derivative for the given joint:
+
+```math
+\\dot{q} = J_{v \\rightarrow \\dot{q}} v
+```
+"""
+function velocity_to_configuration_derivative_jacobian(joint::Joint, q::AbstractVector)
+    @boundscheck check_num_positions(joint, q)
+    velocity_to_configuration_derivative_jacobian(joint.joint_type, q)
+end
+
+"""
+$(SIGNATURES)
+
+See [`velocity_to_configuration_derivative_jacobian`](@ref). 
+
+This method does its computation in-place, storing the result in `J`. 
+"""
+function velocity_to_configuration_derivative_jacobian!(J::AbstractMatrix, joint::Joint, q::AbstractVector)
+    @boundscheck size(J) == (num_positions(joint), num_velocities(joint)) || error("Wrong size")
+    J .= velocity_to_configuration_derivative_jacobian(joint, q)
+end
+
+"""
+$(SIGNATURES)
+
+Compute the jacobian ``J_{\\dot{q} \\rightarrow v}`` which maps joint
+configuration derivative to velocity for the given joint:
+
+```math
+v = J_{\\dot{q} \\rightarrow v} \\dot{q}
+```
+"""
+function configuration_derivative_to_velocity_jacobian(joint::Joint, q::AbstractVector)
+    @boundscheck check_num_positions(joint, q)
+    configuration_derivative_to_velocity_jacobian(joint.joint_type, q)
+end
+
+"""
+$(SIGNATURES)
+
+See [`configuration_derivative_to_velocity_jacobian`](@ref). 
+
+This method does its computation in-place, storing the result in `J`. 
+"""
+function configuration_derivative_to_velocity_jacobian!(J::AbstractMatrix, joint::Joint, q::AbstractVector)
```

Comment:
I think we could get rid of this method entirely. We only *need* the in-place version that acts on the entire MechanismState.